pkg/portal/portal.go

```diff
@@ -455,7 +436,23 @@ func (p *portal) getSubscriptionDocument(ctx context.Context, key string) (*api.
 	return doc, err
 }
 
+func (p *portal) getResourceID(r *http.Request) string {
+	apiVars := mux.Vars(r)
+	subscriptionID := apiVars["subscription"]
+	resourceGroup := apiVars["resourceGroup"]
+	clusterName := apiVars["clusterName"]
+	return strings.ToLower(
+		fmt.Sprintf(
+			"/subscriptions/%s/resourceGroups/%s/providers/Microsoft.RedHatOpenShift/openShiftClusters/%s",
+			subscriptionID, resourceGroup, clusterName))
+}
```

Comment:
I think this function should be changed to receive subscriptionId, resourceGroup and clusterName directly and not worry about getitng out the variables from the request, because the place you're calling `getResourceID` you're already extracting those variables out form apiVars.

Reply:
I actually wrote it to receive subscriptionId, resourceGroup, clusterName only previously but since I saw that most of the places where it is being used, these values are being extracted, so to make once piece of code reusable, I then changed it to take a *http.Request to extract these variables inside the function itself.

Reply:
Hmmm, cause now we're extracting these variables twice, and if others use this code, they'd be forced to validate these same variables outside of this function. My suggestion then is either:
1. Move the apiVars validation code into this function, returning errors or something like that so that the calling function can send the appropriate http error response OR
2. Change it back to accepting only those values without extracting them.

Reply:
I will change it back then.